spec/std/socket/address_spec.cr

```diff
@@ -273,7 +276,11 @@ describe Socket do
     Socket.ip?("1:2:3:4:5:6:7:8::").should be_false
     Socket.ip?("1:2:3:4:5:6:7::9").should be_false
     Socket.ip?("::1:2:3:4:5:6").should be_true
-    Socket.ip?("::1:2:3:4:5:6:7").should be_true
+    {% if flag?(:win32) %}
+      Socket.ip?("::1:2:3:4:5:6:7").should be_false
+    {% else %}
+      Socket.ip?("::1:2:3:4:5:6:7").should be_true
+    {% end %}
```

Comment:
This is a bit weird. Do you know why this differs?

Reply:
No idea, but probably a glitch in the implementation.

There are actually a number of expectations commented out in that spec because they fail with some implementations. I just thought it would be better to test that expectation for POSIX implementations. We could potentially skip the win32 branch because it's faulty. But this at least documents a known failure.
And maybe we could do something about fixing such issues outselves at some point.

Reply:
This now returns true on Windows CI on master since https://github.com/crystal-lang/crystal/runs/5044032259?check_suite_focus=true, which may be due to `windows-latest` upgrading to Windows Server 2022.

Reply:
Oh that's great. However, now we must check for the windows version 🙈  Or just skip this spec on windows...